internal/tofu/context_plan.go

```diff
@@ -526,23 +526,34 @@ func (c *Context) postPlanValidateMoves(config *configs.Config, stmts []refactor
 // All import target addresses with a key must already exist in config.
 // When we are able to generate config for expanded resources, this rule can be
 // relaxed.
-func (c *Context) postPlanValidateImports(config *configs.Config, importTargets []*ImportTarget, allInst instances.Set) tfdiags.Diagnostics {
+func (c *Context) postPlanValidateImports(resolvedImports *ResolvedImports, allInst instances.Set) tfdiags.Diagnostics {
 	var diags tfdiags.Diagnostics
-	for _, it := range importTargets {
+	for resolvedImport := range resolvedImports.imports {
 		// We only care about import target addresses that have a key.
 		// If the address does not have a key, we don't need it to be in config
 		// because are able to generate config.
-		if it.Addr.Resource.Key == nil {
+		address, addrParseDiags := addrs.ParseAbsResourceInstanceStr(resolvedImport.AddrStr)
+		if addrParseDiags.HasErrors() {
+			return addrParseDiags
```

Comment:
Nit: add addrParseDiags to diags and use the next continue instead of checking HasErrors()?

Reply:
I personally don't like that approach much, as it relies on implicit behaviour of `ParseAbsResourceInstanceStr`

Also, there's a change the `if` afterwards will be removed, as it's technically just a way to skip an unnecessary iteration of the for loop, but it's not really needed. Once that `if` is removed, we will lose error handling for `ParseAbsResourceInstanceStr`

Reply:
I actually ended up removing that `if`, as a I made to how config generation works resulted in it skipping validation for some imports when it shouldn't have